diff --git a/node_modules/expo-modules-core/ios/Core/Logging/PersistentFileLog.swift b/node_modules/expo-modules-core/ios/Core/Logging/PersistentFileLog.swift
index 8938578..daa648e 100644
--- a/node_modules/expo-modules-core/ios/Core/Logging/PersistentFileLog.swift
+++ b/node_modules/expo-modules-core/ios/Core/Logging/PersistentFileLog.swift
@@ -2,8 +2,8 @@
 
 import Foundation
 
-public typealias PersistentFileLogFilter = (String) -> Bool
-public typealias PersistentFileLogCompletionHandler = (Error?) -> Void
+public typealias PersistentFileLogFilter = @Sendable (String) -> Bool
+public typealias PersistentFileLogCompletionHandler = @Sendable (Error?) -> Void
 
 /**
  * A thread-safe class for reading and writing line-separated strings to a flat file.
diff --git a/node_modules/expo-modules-core/ios/Core/Views/SwiftUI/SwiftUIHostingView.swift b/node_modules/expo-modules-core/ios/Core/Views/SwiftUI/SwiftUIHostingView.swift
index b1c9f8b..704bf79 100644
--- a/node_modules/expo-modules-core/ios/Core/Views/SwiftUI/SwiftUIHostingView.swift
+++ b/node_modules/expo-modules-core/ios/Core/Views/SwiftUI/SwiftUIHostingView.swift
@@ -42,7 +42,8 @@ extension ExpoSwiftUI {
   /**
    A hosting view that renders a SwiftUI view inside the UIKit view hierarchy.
    */
-  public final class HostingView<Props: ViewProps, ContentView: View<Props>>: ExpoView, @MainActor AnyExpoSwiftUIHostingView {
+  @MainActor
+  public final class HostingView<Props: ViewProps, ContentView: View<Props>>: ExpoView, AnyExpoSwiftUIHostingView {
     /**
      Props object that stores all the props for this particular view.
      It's an environment object that is observed by the content view.
diff --git a/node_modules/expo-modules-core/ios/Core/Views/SwiftUI/SwiftUIVirtualView.swift b/node_modules/expo-modules-core/ios/Core/Views/SwiftUI/SwiftUIVirtualView.swift
index 6d65323..58891ae 100644
--- a/node_modules/expo-modules-core/ios/Core/Views/SwiftUI/SwiftUIVirtualView.swift
+++ b/node_modules/expo-modules-core/ios/Core/Views/SwiftUI/SwiftUIVirtualView.swift
@@ -7,6 +7,7 @@ extension ExpoSwiftUI {
    An NSObject acting as a fake UIView for RCTMountingManager to represent a SwiftUI view.
    This class is the Swift component of SwiftUIVirtualView, as referenced in ExpoFabricView.swift.
    */
+  @MainActor
   final class SwiftUIVirtualView<Props: ViewProps, ContentView: View<Props>>: SwiftUIVirtualViewObjC, ExpoSwiftUIView {
     var uiView: UIView?
     
@@ -193,7 +194,7 @@ extension ExpoSwiftUI {
 
 // MARK: - ViewWrapper
 
-extension ExpoSwiftUI.SwiftUIVirtualView: @MainActor ExpoSwiftUI.ViewWrapper {
+extension ExpoSwiftUI.SwiftUIVirtualView: ExpoSwiftUI.ViewWrapper {
   func getWrappedView() -> Any {
     if let wrapper = contentView as? ExpoSwiftUI.ViewWrapper {
       return wrapper.getWrappedView()
diff --git a/node_modules/expo-modules-core/ios/Core/Views/ViewDefinition.swift b/node_modules/expo-modules-core/ios/Core/Views/ViewDefinition.swift
index 5dc73e3..67b7b8e 100644
--- a/node_modules/expo-modules-core/ios/Core/Views/ViewDefinition.swift
+++ b/node_modules/expo-modules-core/ios/Core/Views/ViewDefinition.swift
@@ -118,7 +118,8 @@ extension ConcurrentFunctionDefinition: ViewDefinitionFunctionElement {
   public typealias ViewType = FirstArgType
 }
 
-extension UIView: @MainActor AnyArgument {
+@MainActor
+extension UIView: AnyArgument {
   public static func getDynamicType() -> AnyDynamicType {
     return DynamicViewType(innerType: Self.self)
   }
diff --git a/node_modules/expo-modules-core/ios/DevTools/URLAuthenticationChallengeForwardSender.swift b/node_modules/expo-modules-core/ios/DevTools/URLAuthenticationChallengeForwardSender.swift
index 8c3efcb..3ea385e 100644
--- a/node_modules/expo-modules-core/ios/DevTools/URLAuthenticationChallengeForwardSender.swift
+++ b/node_modules/expo-modules-core/ios/DevTools/URLAuthenticationChallengeForwardSender.swift
@@ -3,7 +3,7 @@
 /**
  A helper class to forward URLAuthenticationChallenge completion handler to URLAuthenticationChallengeSender
  */
-internal final class URLAuthenticationChallengeForwardSender: NSObject, URLAuthenticationChallengeSender {
+internal final class URLAuthenticationChallengeForwardSender: NSObject, URLAuthenticationChallengeSender, @unchecked Sendable {
   let completionHandler: (URLSession.AuthChallengeDisposition, URLCredential?) -> Void
 
   init(completionHandler: @escaping (URLSession.AuthChallengeDisposition, URLCredential?) -> Void) {
diff --git a/node_modules/expo-modules-core/ios/DevTools/URLSessionSessionDelegateProxy.swift b/node_modules/expo-modules-core/ios/DevTools/URLSessionSessionDelegateProxy.swift
index dbc5b65..1f2c962 100644
--- a/node_modules/expo-modules-core/ios/DevTools/URLSessionSessionDelegateProxy.swift
+++ b/node_modules/expo-modules-core/ios/DevTools/URLSessionSessionDelegateProxy.swift
@@ -3,7 +3,7 @@
 /**
  Shared URLSessionDelegate instance and delete calls back to ExpoRequestInterceptorProtocol instances.
  */
-public final class URLSessionSessionDelegateProxy: NSObject, URLSessionDataDelegate {
+public final class URLSessionSessionDelegateProxy: NSObject, URLSessionDataDelegate, @unchecked Sendable {
   private let dispatchQueue: DispatchQueue
   private var delegateMap: [AnyHashable: URLSessionDataDelegate] = [:]
 
diff --git a/node_modules/expo-modules-core/ios/ReactDelegates/ExpoReactDelegate.swift b/node_modules/expo-modules-core/ios/ReactDelegates/ExpoReactDelegate.swift
index 81871a0..c0ddd60 100644
--- a/node_modules/expo-modules-core/ios/ReactDelegates/ExpoReactDelegate.swift
+++ b/node_modules/expo-modules-core/ios/ReactDelegates/ExpoReactDelegate.swift
@@ -34,6 +34,7 @@ public class ExpoReactDelegate: NSObject {
       .first(where: { _ in true })
   }
 
+  @MainActor
   @objc
   public func createRootViewController() -> UIViewController {
     return self.handlers.lazy
